baseframe/filters.py

```diff
@@ -211,3 +212,57 @@ def longdate(value):
     else:
         dt = value
     return dt.strftime('%e %B %Y')
+
+
+@baseframe.app_template_filter('date')
+def date_filter(value, format='medium', usertz=True):
+    if isinstance(value, datetime):
+        if value.tzinfo is None:
+            if usertz:
+                dt = UTC.localize(value).astimezone(get_timezone())
+            else:
+                dt = value
+        else:
+            if usertz:
+                dt = value.astimezone(get_timezone())
+            else:
+                dt = value
+    else:
+        dt = value
```

Comment:
This could be much simpler:
```python
if isinstance(value, datetime) and usertz:
    if value.tzinfo is None:
        dt = UTC.localize(value).astimezone(get_timezone())
    else:
        dt = value.astimezone(get_timezone())
else:
    dt = value
```

Reply:
This is pending.